config/proposer/loader/loader.go

```diff
@@ -259,18 +262,38 @@ func (psl *settingsLoader) processProposerSettings(loadedSettings, dbSettings *v
 	return newSettings
 }
 
-func processBuilderConfig(current *validatorpb.BuilderConfig, override *validatorpb.BuilderConfig, gasLimitOnly *validator.Uint64) *validatorpb.BuilderConfig {
-	if current != nil {
-		current.GasLimit = reviewGasLimit(current.GasLimit)
-		if override != nil {
-			current.Enabled = override.Enabled
-		}
-		if gasLimitOnly != nil {
-			current.GasLimit = *gasLimitOnly
-		}
+func processBuilderConfig(current *validatorpb.BuilderConfig, options *flagOptions) *validatorpb.BuilderConfig {
+	// If there are no options, return what was passed in
+	if options == nil {
 		return current
 	}
-	return override
+
+	// Initialize an override variable
+	var override *validatorpb.BuilderConfig
+	if options.builderConfig != nil {
+		// Convert the builder config to consensus form if it exists
+		override = options.builderConfig.ToConsensus()
+	}
+	// If there's nothing to process further, return the override or current based on what's available
+	if current == nil {
+		return override
```

Comment:
This will return nil if both `options.builderConfig` and `current` are nil. Is this OK?

Reply:
yeah it means there's nothing to override